docs/tools/frequently-asked-questions.md

```diff
@@ -0,0 +1,80 @@
+---
+id: frequently-asked-questions
+title: Frequently Asked Questions 
+sidebar_label: FAQ 
+---
+# Need Some Help?
+
+- [What is a SSH (Secure Shell) ?](#what-is-a-ssh-(secure-shell))
+- [How to generate a new SSH key?](#how-to-generate-a-new-ssh-key)
+- [How to add your SSH key to the ssh-agent](#how-to-add-your-ssh-key-to-the-ssh-agent)
+- [What is the difference between a public and private key?](#what-is-the-difference-between-a-public-and-private-key)
+
+
+## What is a SSH (Secure Shell) ?
+
+The `SSH` is also known as Secure Shell or Secure Socket Shell, it is a protocol that you can use to connect and aunthenticate to remote servers and services. [More About SSH in GitHub](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/about-ssh)
+
+
+## How to generate a new SSH key?
+
+This guide is for Windows users , more information for Linux and Mac [on this site](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)
+
+- Open Git Bash
+
+- Paste the text below, substitute `"your_email@example.com"` with your GitHub email address.
+```coffee
+$ ssh-keygen -t ed25519 -C "your_email@example.com"
+```
+---
+>**NOTE :** 
+If you are using a legacy system that doesn't support the `ed25519` algorithm, use:
+
+```coffee
+$ ssh-keygen -t rsa -b 4096 -C "your_email@example.com"
+```
+---
+
+- It will ask for a file to save the key , just press enter. This accepts the default file location. 
+
+- At the prompt , type a secure passphrase. For security reasons it should be as complex as possible.
+
+Now you have a new SSH key!
+
+## How to add your SSH key to the ssh-agent
+
+- Ensure the ssh-agent is running [more information](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/working-with-ssh-key-passphrases) , or start it manually:
+
+```coffee
```

Comment:
I like how you use a code block here but instead of `coffee` script I think we can use `shell` script instead